espresso/environment/1_espresso_benchmark_test.go

```diff
@@ -0,0 +1,153 @@
+	}
+
+	defer env.Stop(t, system)
+	defer env.Stop(t, espressoDevNode)
+
+	caffNode, err := env.LaunchDecaffNode(t, system, espressoDevNode)
+	if have, want := err, error(nil); have != want {
+		t.Fatalf("failed to start caff node:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"\n", have, want)
+	}
+
+	// Shut down the Caff Node
+	defer env.Stop(t, caffNode)
+
+	keys := system.Cfg.Secrets
+	addresses := keys.Addresses()
+
+	signer := geth_types.LatestSignerForChainID(system.Cfg.L2ChainIDBig())
+
+	// Submit Transactions to the Sequencer
+	bencher := benchmark.CreateBenchmarker(
+		ctx,
+		benchmark.WithSeqClient(system.NodeClient(e2esys.RoleSeq)),
+		benchmark.WithCaffClient(system.NodeClient(env.RoleCaffNode)),
+		benchmark.WithVerifyClient(system.NodeClient(e2esys.RoleVerif)),
+		benchmark.AddSubmitter(benchmark.BenchmarkSubmitterConfig{
+			Interval: 100 * time.Millisecond,
+			To:       &addresses.Bob,
+			Value:    big.NewInt(1),
+			Signer:   signer,
+			ChainID:  system.RollupConfig.L2ChainID,
+			Key:      keys.Alice,
+		}),
+		benchmark.AddSubmitter(benchmark.BenchmarkSubmitterConfig{
+			Interval: 100 * time.Millisecond,
+			To:       &addresses.Mallory,
+			Value:    big.NewInt(1),
+			Signer:   signer,
+			ChainID:  system.RollupConfig.L2ChainID,
+			Key:      keys.Bob,
+		}),
+		benchmark.AddSubmitter(benchmark.BenchmarkSubmitterConfig{
+			Interval: 100 * time.Millisecond,
+			To:       &addresses.Alice,
+			Value:    big.NewInt(1),
+			Signer:   signer,
+			ChainID:  system.RollupConfig.L2ChainID,
+			Key:      keys.Mallory,
+		}),
+	)
+
+	// Alright, let's run the benchmark for a couple of minutes
+
+	{
+		ctx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+		defer cancel()
+
+		stats, err := bencher.RunWithContext(ctx)
+		if have, want := err, error(nil); have != want {
+			t.Fatalf("failed to run benchmark:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"\n", have, want)
+		}
+
+		// Let's check the statistics
+		metrics := benchmark.ComputeRawMetricsStatistics(stats)
+
+		if have, want := metrics.SubmittedToReceipt.Count, 0; have <= want {
+			t.Errorf("expected to have a positive count for receipts received:\nhave:\n\t\"%v\"\nwant:\n\t> \"%v\"\n", have, want)
+		}
+
+		if have, want := metrics.ReceiptToCaff.Count, 0; have <= want {
+			t.Errorf("expected to have a positive count for caff headers:\nhave:\n\t\"%v\"\nwant:\n\t> \"%v\"\n", have, want)
+		}
+
+		if have, want := metrics.ReceiptToVerify.Count, 0; have <= want {
+			t.Errorf("expected to have a positive count for verify headers:\nhave:\n\t\"%v\"\nwant:\n\t> \"%v\"\n", have, want)
+		}
+
+		// We do not expect a signification amount of variance or std deviation
+		if have, want := metrics.SubmittedToReceipt.StdDev, 2*time.Second; have > want {
+			t.Errorf("expected a small amount of variance in the submitted to receipt time:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"\n", have, want)
+		}
+
+		if have, want := metrics.ReceiptToCaff.StdDev, 2*time.Second; have > want {
```

Comment:
Should not it take more time to get the receipt from the Caff node, like around 4s?

Reply:
I do think that it might take longer than two seconds to get a verification from the Caff Node for a response.  However, in this case, we're not actually comparing the time it took for the Caff Node to provide a response.

Instead, we're looking at how much the time taken to get that response varies from the mean time to get the response.  The lower than number, the lower the variance and the higher the consistency.

Reply:
Oh I missed `StdDev`. Thanks!